### PR TITLE
Bugfixes from develop

### PR DESCRIPTION
**Pull request by dimas1185**
## Change Description

back port of:
https://github.com/EOSIO/eos/pull/10590
https://github.com/EOSIO/eos/pull/10584
https://github.com/EOSIO/eos/pull/10580

Fixes are:
* `net_plugin` bugfix for some race conditions when nodes stop syncing
* `test_control_plugin` fix for condition when producer produces less than 12 blocks in round (due to performance issues).
* test framework fixes: increased transaction expiration in `Node.transferFunds` and blocks range in `Node.getBlockNumByTransId`

Integration tests where those bugs were found:
* terminate-scenarios-test-replay
* nodeos_short_fork_take_over_lr_test
* restart-scenarios-test-resync
* distributed-transactions-test

## Change Type
**Select *ONE*:**
- [ ] Documentation

- [x] Stability bug fix

- [ ] Other

- [ ] Other - special case

## Testing Changes
**Select *ANY* that apply:**
- [ ] New Tests

- [ ] Existing Tests

- [x] Test Framework

- [ ] CI System

- [ ] Other

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

**Comment by heifner** [comment]
Can you summarize the PRs in the description so someone doesn't have to click through each one to know what it is about.

**Comment by dimas1185** [comment]
> Can you summarize the PRs in the description so someone doesn't have to click through each one to know what it is about.

@heifner  done

**Review by heifner — APPROVED** [review]
